File: max-consecutive-ones/max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = 0
        max_count = 0
        
        for num in nums:
            if num == 1:
                count += 1
                max_count =  max(count, max_count)
            else:
                count = 0
        return max_count

File: max-consecutive-ones/max-consecutive-ones.rb
# @param {Integer[]} nums
# @return {Integer}
def find_max_consecutive_ones(nums)
    max, count = 0,0
    
    nums.each do |num|
        if num == 1
            count += 1
            max = count if count > max
        else
            count = 0
        end
    end
    max
end